server/src/main/java/org/elasticsearch/cluster/metadata/Template.java

```diff
@@ -291,4 +291,62 @@ static boolean mappingsEquals(CompressedXContent m1, CompressedXContent m2) {
         );
         return Maps.deepEquals(thisUncompressedMapping, otherUncompressedMapping);
     }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder builder(@Nullable Template template) {
```

Comment:
The `@Nullable` here surprised me a bit. It wasn't immediately obvious what calling `builder(null)` would do. Is this here just for the convenience of a single caller, or did I miss some? I wonder if it would be better to make this parameter required, and do the `?:` in that caller.

Alternatively, keep the `@Nullable` but add a one-line javadoc.

Reply:
It is a convenience method for a singler caller indeed. I thought that there might be more occurrences of this so it would be useful. Since we do not have more cases I am happy to remove it and we can evaluate this in the future.